src/classes/MyHandler.ts

```diff
@@ -238,12 +237,12 @@ export = class MyHandler extends Handler {
     onGroupRelationship(groupID: SteamID, relationship: number): void {
         log.debug('Group relation changed', { steamID: groupID, relationship: relationship });
         if (relationship === SteamUser.EClanRelationship.Invited) {
-            const join = !this.groups.includes(groupID.getSteamID64());
+            const join = this.groups.includes(groupID.getSteamID64());
 
             log.info(
                 'Got invited to group ' + groupID.getSteamID64() + ', ' + (join ? 'accepting...' : 'declining...')
             );
-            this.bot.client.respondToGroupInvite(groupID, !this.groups.includes(groupID.getSteamID64()));
+            this.bot.client.respondToGroupInvite(groupID, this.groups.includes(groupID.getSteamID64()));
```

Comment:
use the `join` constant you have just created instead of `this.groups.includes(groupID.getSteamID64())`

Reply:
also maybe don't call the variable `join`, but something along the lines of `shouldJoin` to confuse possible confusion.